### ISSUE TITLE
ISIS Reflectometry: Prevent external table modification while processing

### ISSUE THREAD

**Issue by PranavBahuguna** [mantidproject/mantid#19951]
While the current interface greys out the `process` buttons and prevents window closure while processing runs, modification to the table of data is not prevented and can cause errors when done. These include:

- Adding/deleting rows/groups
- Grouping rows
- Cutting/pasting rows
- Editing row data
- Transferring rows

In addition, the whole `Reflectometry` menu (except for `Export .TBL` and `Slit Calculator`) should be disabled (#21215). Furthermore, one should not be able to select any run in the table while processing occurs.

Edit: Another bug I want to address in this is that currently, pausing reduction will immediately make the `process` button immediately available (before reduction itself has actually paused). If `process` is clicked before reduction has finished, a critical error would be thrown. `Process` should be re-enabled only when reduction has finished, not just when the pause button is clicked. This has been noted as part of #21167.

In summary we need to:
- [x] Disable editing rows in the data processor widget
- [x] Disable relevant items in the menus #21215

**Comment by gemmaguest** [comment]
I'm closing this issue as the main fixes have been added. There is one related issue which is not a priority at the moment so has been moved to #21167
